### PR TITLE
feat(Kickassanime): Added anime posters as LargeImageKey

### PR DESCRIPTION
**Pull request by StrikerFRFX**
Added anime posters to be displayed on 2 different pages:

Img1 
> Displays when watching an episode 

Img2
> Displays when viewing page of an anime

Both are toggleable and if the image is unresolveable for the poster, defaults to the KAA Logo.

### CODE REVIEW

File: websites/K/KickAssAnime/presence.ts [hunk 1]
```diff
@@ -79,11 +79,10 @@ presence.on(
...
-  const presenceData: PresenceData = {
-      largeImageKey: "kaa"
-    },
+  const presenceData: PresenceData = { largeImageKey: "kaa" },
     buttons = await presence.getSetting("buttons"),
-    newLang = await presence.getSetting("lang");
+    newLang = await presence.getSetting("lang"),
+    cover = await presence.getSetting("cover");
 
   presenceData.startTimestamp = browsingTimestamp;
```
Comment (EncryptedDev) [inline websites/K/KickAssAnime/presence.ts:87]: Set this in the presenceData obj

File: websites/K/KickAssAnime/presence.ts [hunk 2]
```diff
@@ -79,11 +79,10 @@ presence.on(
...
 
 presence.on("UpdateData", async () => {
-  const presenceData: PresenceData = {
-      largeImageKey: "kaa"
-    },
+  const presenceData: PresenceData = { largeImageKey: "kaa" },
     buttons = await presence.getSetting("buttons"),
-    newLang = await presence.getSetting("lang");
+    newLang = await presence.getSetting("lang"),
+    cover = await presence.getSetting("cover");
```
Comment (EncryptedDev) [inline websites/K/KickAssAnime/presence.ts:85]: Promise.all?

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
bump version too

File: websites/K/KickAssAnime/presence.ts [hunk 3]
```diff
@@ -211,6 +219,14 @@ presence.on("UpdateData", async () => {
...
     presenceData.state = `${currentAnimeTitle}`;
     presenceData.smallImageKey = "searching";
+    presenceData.largeImageKey = cover
+      ? document
+          .querySelector<HTMLElement>("div.poster")
+          .style.backgroundImage.replace("url(", "")
+          .replace(")", "")
+          .replace('"', "")
+          .replace('"', "")
+      : "kaa";
```
Comment (Bas950) [inline websites/K/KickAssAnime/presence.ts:229]:
```suggestion
    presenceData.largeImageKey = cover
      ? document
          .querySelector<HTMLElement>("div.poster")
          .style.backgroundImage.match(/"(.*)"/)[1]
      : "kaa";
```

File: websites/K/KickAssAnime/presence.ts [hunk 4]
```diff
@@ -129,6 +128,15 @@ presence.on("UpdateData", async () => {
...
         }
+
+        presenceData.largeImageKey = cover
+          ? document
+              .querySelector<HTMLElement>("div.info-header.p-3.px-2")
+              .style.backgroundImage.replace("url(", "")
+              .replace(")", "")
+              .replace('"', "")
+              .replace('"', "")
+          : "kaa";
```
Comment (Bas950) [inline websites/K/KickAssAnime/presence.ts:139]:
```suggestion
        presenceData.largeImageKey = cover
          ? document
              .querySelector<HTMLElement>("div.info-header.p-3.px-2")
              .style.backgroundImage.match(/"(.*)"/)[1]
          : "kaa";
```

**Review by Bas950 — DISMISSED** [review]
(test it tho)

File: websites/K/KickAssAnime/presence.ts (hunk 2 above)
Comment (StrikerFRFX) [reply websites/K/KickAssAnime/presence.ts:85]: what

File: websites/K/KickAssAnime/presence.ts (hunk 1 above)
Comment (StrikerFRFX) [reply websites/K/KickAssAnime/presence.ts:87]: what

**Comment by StrikerFRFX** [comment]
image
image

Proof of still working after changes ^

**Review by Slowlife01 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**